avm/res/network/virtual-network-gateway/main.bicep

```diff
@@ -79,6 +79,15 @@ param enableBgp bool = true
 @description('Optional. ASN value.')
 param asn int = 65815
 
+@description('Optional. Primary custom Azure APIPA BGP IP addresses. Azure supports BGP IP in the ranges 169.254.21.* and 169.254.22.*. This is only used if bgp is enabled.')
+param primaryCustomBgpIPs array = [] 
+
+@description('Optional. Secondary custom Azure APIPA BGP IP addresses. Azure supports BGP IP in the ranges 169.254.21.* and 169.254.22.*. This is only used for active-active configuration if bgp is enabled.')
+param secondaryCustomBgpIPs array = [] 
+
+@description('Optional. The weight added to routes learned from this BGP speaker.')
```

Comment:
We have a lot of parameters used by the bgpSettings block. Also I sense variable implementation is becoming complex for this one. I wonder if it's not time to perhaps leverage User-Defined-Types for the bgpSettings? Following the API implementation here https://learn.microsoft.com/en-us/azure/templates/microsoft.network/virtualnetworkgateways?pivots=deployment-language-bicep#bgpsettings

That would help decreasing the number of input parameters while also improving the consumption experience. If there are no blockers to leverage that I'd strongly suggest to implement it in that way

Reply:
This is being discussed offline, next step is on me

Reply:
Hey @eriqua, is this still active? :)

Reply:
Yes it is, will get there, I promise 😄

Reply:
All good 😄 Just making sure

Reply:
Discussed offline. Resolving this comment and keeping the other one open